### PR TITLE
Rocmsmidocs reorg

### PR DESCRIPTION
**Pull request by Rmalavally**

### CODE REVIEW

File: docs/install/install.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  :keywords: install, SMI, library, api, AMD, ROCm
+
+
+*********************
+Installing ROCm SMI
+*********************
+
+Planned Deprication Notice 
+ROCm System Management Interface (ROCm SMI) Library is planned to be ***depricated***. Release date to be announced soon. Please start migrating to AMD SMI.
+  - Documentation: [https://rocm.docs.amd.com](https://rocm.docs.amd.com/projects/amdsmi/en/latest/)
```
Comment (samjwu) [inline docs/install/install.rst]: The doc build failure (https://readthedocs.com/projects/advanced-micro-devices-rocm-smi-lib/builds/2207319/) flags this line:
```
/home/docs/checkouts/readthedocs.org/user_builds/advanced-micro-devices-rocm-smi-lib/checkouts/168/docs/install/install.rst:12: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/advanced-micro-devices-rocm-smi-lib/checkouts/168/docs/install/install.rst:40: ERROR: Content block expected for the "Note" directive; none found.
```

This is Markdown syntax

Comment (samjwu) [inline docs/install/install.rst]:
```suggestion
Documentation: `https://rocm.docs.amd.com <https://rocm.docs.amd.com/projects/amdsmi/en/latest/>`_
```

File: docs/install/install.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  - Github: [https://github.com/ROCm/amdsmi](https://github.com/ROCm/amdsmi)
```
Comment (samjwu) [inline docs/install/install.rst]:
```suggestion
Github: `https://github.com/ROCm/amdsmi <https://github.com/ROCm/amdsmi>`_
```

File: docs/install/install.rst (hunk 1 above)
Comment (samjwu) [reply docs/install/install.rst]: RST syntax for links: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html

Comment (Rmalavally) [reply docs/install/install.rst]: Thanks, @samjwu. Markdown syntax has not caused a build error or failure before. It just displays incorrectly in the file. I have retained a link in Markdown syntax in index.rst, and the build works fine. 
markdownlink1

File: docs/sphinx/_toc.yml.in [hunk 3]
```diff
@@ -5,27 +5,38 @@ defaults:
...
-    title: C++ Tutorials
-  - file: python_tutorials
-    title: Python Tutorials
-- caption: How to Guide
+  - file: install/install
+    title: ROCm SMI installation
+ 
+- caption: How to 
   entries:
   - file: c++_usage
```
Comment (samjwu) [inline docs/sphinx/_toc.yml.in]:
```suggestion
  - file: how-to/use-cpp
```

Comment (Rmalavally) [reply docs/sphinx/_toc.yml.in]: Fixed.

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
+.. meta::
+  :description: ROCm SMI
+  :keywords: install, SMI, library, api, AMD, ROCm
+
+****************************************************
+ROCm System Management Interface (ROCm SMI) library
+****************************************************
+
+The ROCm System Management Interface library, or ROCm SMI library, is part of the ROCm software stack. It is a C library for Linux that provides a user space interface for applications to monitor and control GPU applications.
+
```
Comment (lpaoletti) [inline docs/index.rst:10]: Perhaps we should include a cross-reference to AMD SMI and explain the difference. 
And we should do the same for AMD SMI main page.

**Review by lpaoletti — COMMENTED** [review]
These How to titles don't line up:
image
image

File: docs/reference/python_api.rst [hunk 5]
```diff
@@ -1,8 +1,8 @@
-====================
-Python API Reference
-====================
+=====================
+Python API reference
```
Comment (lpaoletti) [inline docs/reference/python_api.rst:2]: This topic is empty

image

**Review by lpaoletti — COMMENTED** [review]
Why is there only one file in the Reference folder?
image

**Comment by Rmalavally** [comment]
> Why is there only one file in the Reference folder? image

We are linking to the Doxygen-generated HTML files directly from the toc.yml.in file. This is the additional file from the Component team. I have reached out to the Component team for help. This content is not rendered on the ROCm docs portal even today.
https://rocm.docs.amd.com/projects/rocm_smi_lib/en/latest/python_api.html

File: docs/reference/python_api.rst (hunk 5 above)
Comment (Rmalavally) [reply docs/reference/python_api.rst:2]: Yes. I have reached out to the Component team for clarity. The existing page on the ROCm docs portal is empty as well.

https://rocm.docs.amd.com/projects/rocm_smi_lib/en/latest/python_api.html

**Comment by Rmalavally** [comment]
> These How to titles don't line up: image image

Fixed.

**Comment by Rmalavally** [comment]
> Why is there only one file in the Reference folder? image

Fixed.

**Comment by Rmalavally** [comment]
> These How to titles don't line up: image image

Fixed.

File: docs/index.rst (hunk 4 above)
Comment (Rmalavally) [reply docs/index.rst:10]: This will require reaching out to the Component team. Please advise.

**Review by lpaoletti — APPROVED** [review]

**Pull request closed without merging**